Upstream-Status: Backport

This backports a patch from openjdk 9 repository that fixes undefined behaviour
where address of a temporary was returned.

Original patch at hg.openjdk.java.net/jdk9/dev/hotspot/rev/6217fd2c767b
by Severin Gehwolf

Signed-off-by: Erkka Kääriä <erkka.kaaria@intel.com>

--- openjdk/hotspot/src/os_cpu/linux_zero/vm/os_linux_zero.cpp
+++ openjdk/hotspot/src/os_cpu/linux_zero/vm/os_linux_zero.cpp
@@ -61,8 +61,8 @@
 #endif
 
 address os::current_stack_pointer() {
-  address dummy = (address) &dummy;
-  return dummy;
+  // return the address of the current function
+  return (address)__builtin_frame_address(0);
 }
 
 frame os::get_sender_for_C_frame(frame* fr) {
-- 
2.1.4

